Book-A-Desk.Api/EmailNotification.fs

```diff
@@ -0,0 +1,46 @@
+namespace Book_A_Desk.Api
+
+open System
+open System.Net.Mail
+open Book_A_Desk.Api.Models
+open Book_A_Desk.Domain.QueriesHandler
+open Book_A_Desk.Domain.Office.Domain
+
+type EmailNotification =
+    {
+        SendEmailNotification: Booking -> unit
+    }
+
+module EmailNotification =
+    let initialize getEmailServiceConfiguration getOffices =         
+        let sendEmailNotification (booking: Booking) =
+            let config = getEmailServiceConfiguration()        
+            let officeName =
+                let result = OfficeQueriesHandler.getAll getOffices
+                let officeReference = Guid.Parse(booking.Office.Id) |> OfficeId
+                match result with
+                | Ok offices ->
+                    offices
+                    |> List.find (fun (o:Office) -> officeReference.Equals(o.Id))
+                    |> (fun (o:Office) -> 
+                        let (CityName cityName) = o.City
+                        cityName)
+                | Error e ->
+                    "Unknown"
+                    
+            use smtpClient = new SmtpClient(config.SmtpClientUrl)
```

Comment:
Question:
Is SmtpClient intended to be created every time we send an email or to be used as a long living service? If it's meant to be used as a long living service, you could pass in smtp client instead of the configuration, create it once in Program.fs (or maybe in some kind of compositionroot, up to you)

Reply:
As this is a temporary solution anyway and as we send really just a few emails for me this is acceptable for now.
In the future when this is handled by an explicit event handler we should do differently.

Reply:
I'll disagree a little bit with this. I've seen some clients that were not multi-threaded (hence, you need 1 new client per thread) and some that could not handle multiple clients properly (hence, you need a single one).

We should check in which category this client belongs to, just in case it doesn't support multiple clients being used at once.

Reply:
To Do: Initialise the SmtpClient instance and provide this to the sendEmail to reuse the instance.